consul_exporter_test.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import "testing"
+
+func TestNewExporter(t *testing.T) {
+	cases := []struct {
+		uri string
+		ok  bool
+	}{
+		{uri: "", ok: false},
+		{uri: "localhost:8500", ok: true},
+		{uri: "https://localhost:8500", ok: true},
+		{uri: "http://some.where:8500", ok: true},
+		{uri: "fuuuu://localhost:8500", ok: false},
+	}
+
+	for _, test := range cases {
+		_, err := NewExporter(test.uri, "", ".*", true)
+		if test.ok && err != nil {
+			t.Errorf("expected no error w/ %s but got %s", test.uri, err)
+		}
```

Comment:
We need another condition here for `!test.ok && err == nil` to cover all failure cases.